provider/mesh/state.go

```diff
@@ -70,6 +52,47 @@ func (st *notificationState) gc(retention time.Duration) {
 	}
 }
 
+func (st *notificationState) snapshot(w io.Writer) error {
```

Comment:
Shouldn't this use the same codepath as the `Encode` method? The latter writes to a byte slice, and this to a writer. But `Encode` could call this method with the bytes.Buffer as w.

However, `Encode` gobs the whole map, while this method sequentially goes through the map and encodes key and value separately. Is there a reason for that?